src/uv-common.c

```diff
@@ -155,6 +155,82 @@ static const char* uv__unknown_err_code(int err) {
   return copy != NULL ? copy : "Unknown system error";
 }
 
+int uv_translate_posix_error(int sys_errno) {
+  switch (sys_errno) {
```

Comment:
you don't need to do all this, on Unix just return -sys_errno, since that's how we map errors.